tests/php/ORM/DataListEagerLoadingTest.php

```diff
@@ -0,0 +1,967 @@
+            ManyManyEagerLoadObject::class,
+            ManyManySubEagerLoadObject::class,
+            ManyManySubSubEagerLoadObject::class,
+            ManyManyThroughEagerLoadObject::class,
+            ManyManyThroughSubEagerLoadObject::class,
+            ManyManyThroughSubSubEagerLoadObject::class,
+            EagerLoadObjectManyManyThroughEagerLoadObject::class,
+            ManyManyThroughEagerLoadObjectManyManyThroughSubEagerLoadObject::class,
+            ManyManyThroughSubEagerLoadObjectManyManyThroughSubSubEagerLoadObject::class,
+            BelongsManyManyEagerLoadObject::class,
+            BelongsManyManySubEagerLoadObject::class,
+            BelongsManyManySubSubEagerLoadObject::class,
+            MixedHasManyEagerLoadObject::class,
+            MixedHasOneEagerLoadObject::class,
+            MixedManyManyEagerLoadObject::class,
+        ];
+    }
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        // Set non-zero auto increment offset for each object type so we don't end up with the same IDs across
+        // the board. If all of the IDs are 0, 1, 2 then we have no way of knowing if we're accidentally mixing
+        // up relation ID lists between different relation lists for different classes.
+        $schema = DataObject::getSchema();
+        /** @var DataObject $class */
+        $numManyManyTables = 0;
+        foreach (static::getExtraDataObjects() as $i => $class) {
+            $autoIncrementStart = ($i + 1) * 100;
+            $table = $schema->baseDataTable($class);
+            DB::query("ALTER TABLE $table AUTO_INCREMENT = $autoIncrementStart;");
+
+            // Also adjust auto increment for join tables for the same reason.
+            foreach (array_keys($class::config()->get('many_many') ?? []) as $relationName) {
+                $manyManyComponent = $schema->manyManyComponent($class, $relationName);
+
+                $joinTable = $manyManyComponent['join'];
+                if (is_a($manyManyComponent['relationClass'], ManyManyThroughList::class, true)) {
+                    $joinTable = $schema->baseDataTable($joinTable);
+                }
+                $joinAutoIncrementStart = ($numManyManyTables + 1) * 100 + 50;
+                DB::query("ALTER TABLE $joinTable AUTO_INCREMENT = $joinAutoIncrementStart;");
+                $numManyManyTables++;
+            }
+        }
+    }
+
+    /**
+     * @dataProvider provideEagerLoadRelations
+     */
+    public function testEagerLoadRelations(string $iden, array $eagerLoad, int $expected): void
+    {
+        $this->createEagerLoadData();
+
+        $dataList = EagerLoadObject::get()->eagerLoad(...$eagerLoad);
+        list($results, $selectCount) = $this->iterateEagerLoadData($dataList);
+        // We can rely on the non-eager-loaded data being correct, since it's validated by other unit tests
+        list($expectedResults, $_) = $this->iterateEagerLoadData(EagerLoadObject::get());
```

Comment:
This just seems wrong.  We shouldn't be assuming that whatever we read from the database is "expected".  Expected is something that's defined statically like it previously was, please put old expected values back in like they were here https://github.com/silverstripe/silverstripe-framework/pull/10862/files#diff-c4cd01ab13d5b82821b60d1a794a6611cae7d6e9bdd66451151484d8aace7b45L2645

Reply:
We're just testing that the eager loading functionality itself is working, yeah? And we _know_ the lazy loading functionality works and gives us correct results because there's a whole bunch of tests for that already. So we can trust that if the eager loaded list matches the lazy loaded list, it's going to be correct.

That said, for the sake of avoiding ping pong over a unit test which ultimately whether it's static data or correct dynamic data it ends out the same, I'll just go ahead and make this change.
I'll leave the new tests as they are though.